### ISSUE TITLE
Feature Request: Bash/Zsh Tab Completion

### ISSUE THREAD

**Issue by cedricmckinnie** [loft-sh/devspace#1114]
**Is your feature request related to a problem?**  
No, it's more of a potential enhancement

**Which solution do you suggest?**  

_TLDR: Tab Completions are great. Cobra can generate them_

Tab completion can be quite useful as a CLI grows. Support for at least Bash and Zsh would be a good idea.

This might be helpful particularly for the "devspace commands" directive and other directives that require the user to run a separate command like "devspace list" before completing the command that they may be working on. 

For an example of a mature CLI that does this, see the Kubectl CLI.  

It might also be helpful to collect data using "kubectl get" to complete commands that need to gather information from the current context.

Examples

1. "devspace <Tab>" should display the output of "devspace -h" or at least the name of the commands that can be executed.

2. "devspace list <Tab>" should display the output of "devspace list -h" or at least the name of the subcommands that can be executed.
3. "devspace enter <Tab>" should display the output the list of pods available to enter.

I believe Cobra can generate shell completions automatically:

> https://github.com/spf13/cobra/blob/master/bash_completions.md
> https://github.com/spf13/cobra/blob/master/zsh_completions.md

**Which alternative solutions exist?**  

Manually building tab completions

**Additional context**  

/kind feature

**Comment by FabianKramm** [comment]
@cedricmckinnie thanks for creating this issue! Yes sounds like a good idea, I know the bash completions from kubectl and didn't know cobra could generate them automatically. We can definitely add this

**Comment by carlmontanari** [comment]
Closing -- resolved in #1116!
